app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { memo } from 'react';
+import NetworkModals from '../../../../../UI/NetworkModal';
+import { View, TouchableOpacity } from 'react-native';
+import { useSelector } from 'react-redux';
+import ImageIcons from '../../../../../UI/ImageIcon';
+import WarningIcon from 'react-native-vector-icons/FontAwesome';
+import CustomText from '../../../../../Base/Text';
+import EmptyPopularList from '../emptyList';
+import { useNavigation } from '@react-navigation/native';
+import { strings } from '../../../../../../../locales/i18n';
+import { useTheme } from '../../../../../../util/theme';
+import PopularList from '../../../../../../util/networks/customNetworks';
+import createStyles from '../styles';
+import { CustomNetworkProps, Network } from './CustomNetwork.types';
+
+const CustomNetwork = ({
+  isNetworkModalVisible,
+  closeNetworkModal,
+  selectedNetwork,
+  toggleWarningModal,
+  toggleNetworkModal,
+  switchTab,
+}: CustomNetworkProps) => {
+  const savedNetworkList = useSelector(
+    (state: any) =>
+      state.engine.backgroundState.PreferencesController.frequentRpcList,
+  );
+
+  const supportedNetworkList = PopularList.map((network: Network) => {
+    const isAdded = savedNetworkList.some(
+      (savedNetwork: any) => savedNetwork.chainId === network.chainId,
+    );
+    return {
+      ...network,
+      isAdded,
+    };
+  });
+
+  const navigation = useNavigation();
+  const { colors } = useTheme();
+  const styles = createStyles(colors);
+  const filteredPopularList = supportedNetworkList.filter(
+    (val) =>
+      !savedNetworkList.some(
+        (key: { chainId: string }) => val.chainId === key.chainId,
+      ),
+  );
```

Comment:
I think you are already adding a custom property to flag the added network, you can leverage it to optimize the code:
```suggestion
  const filteredPopularList = supportedNetworkList.filter((n) => !n.isAdded) 
```